dali/kernels/test/mat2tensor.h

```diff
@@ -64,6 +66,30 @@ TensorView<StorageCPU, T, ndim> view_as_tensor(cv::Mat &mat) {
   return { mat.ptr<T>(0), tensor_shape<ndim>(mat) };
 }
 
+
+template<typename T = uint8_t, int ndims = 3>
+std::pair<TensorView<StorageGPU, T, ndims>, memory::KernelUniquePtr<T>>
+view_as_tensor_gpu(const cv::Mat &mat) {
+  auto tvcpu = kernels::view_as_tensor<const T, ndims>(mat);
+  auto mem = kernels::memory::alloc_unique<T>(kernels::AllocType::Unified,
+                                              mat.cols * mat.rows * mat.channels());
+  auto tvgpu = kernels::make_tensor_gpu<ndims>(mem.get(), {mat.rows, mat.cols, mat.channels()});
+  kernels::copy(tvgpu, tvcpu);
+  return std::make_pair(tvgpu, std::move(mem));
+}
+
+
+template<typename T = uint8_t, int ndims = 3>
+std::pair<TensorView<StorageGPU, T, ndims>, memory::KernelUniquePtr<T>>
+view_as_tensor_gpu(cv::Mat &mat) {
+  auto tvcpu = kernels::view_as_tensor<T, ndims>(mat);
+  auto mem = kernels::memory::alloc_unique<T>(kernels::AllocType::Unified,
```

Comment:
```suggestion
  auto mem = kernels::memory::alloc_unique<T>(kernels::AllocType::GPU,
```
Don't use unified memory where you don't have to. I know it's for testing only (for now), there it mat actually hide bugs, such as trying to access GPU memory from host or cross-device accesses.